operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Context.java

```diff
@@ -6,5 +6,9 @@
 
   Optional<RetryInfo> getRetryInfo();
 
-  <T> T getSecondaryResource(Class<T> expectedType, String... qualifier);
+  default <T> T getSecondaryResource(Class<T> expectedType) {
```

Comment:
@metacosm is this ok with you, kind would like it to limit it in this scope. Since more qualifiers does not make sense, and if dependent resources will use the previous way it will be backwards compatible anyway. So it can be changed back without breaking the api.